pkg/router/template/util/rewritetarget/rewritetarget_test.go

```diff
@@ -0,0 +1,94 @@
+package rewritetarget_test
+
+import (
+	"github.com/openshift/router/pkg/router/template/util/rewritetarget"
+	"testing"
+)
+
+func Test_SanitizeInput(t *testing.T) {
+	testCases := []struct {
+		name   string
+		input  string
+		output string
+	}{
+		{
+			name:   "single percent should be doubled",
+			input:  `%foo`,
+			output: `%%foo\1`,
+		},
+		{
+			name:   "double percent should be not changed",
+			input:  `%%foo`,
+			output: `%%foo\1`,
+		},
+		{
+			name:   "triple percent should be doubled entirely",
+			input:  `%%%foo`,
```

Comment:
So here we think that the user put "3 percent signs without escaping" and not "1 escaped percent sign plus one unescaped", right?

Reply:
Another interpretation of `%%%` input could be as a combination of `%%` + `%` inputs. Therefore, by default, we assume the user's intent is to do the right escaping (like in case of `%%` input) and we do the minimum necessary to ensure the correct syntax(== `%%%%`).

Reply:
Ah yes it's all coming back to me now. You are correct, we could assume mixed intention for escaped & unescaped, taking each `%` by itself. However, the doubling `%%%` to `%%%%%%` then gets interpreted back to `%%%` again, which is exactly what the author specified in the first place, keeping some consistency.